satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,503 @@
+
+        composites:
+
+          parallax_corrected_VIS006:
+            compositor: !!python/name:satpy.composites.SingleBandCompositor
+            prerequisites:
+              - name: VIS006
+                modifiers: [parallax_corrected]
+
+    Here, ``ctth_alti`` is CTH provided by the ``nwcsaf-geo`` reader, so to use it
+    one would have to pass both on scene creation::
+
+        sc = Scene({"seviri_l1b_hrit": files_l1b, "nwcsaf-geo": files_l2})
+        sc.load(["parallax_corrected_VIS006"])
+
+    Alternately, you can use the lower-level API directly with the
+    :class:`ParallaxCorrection` class, which may be more efficient if multiple
+    datasets need to be corrected.  RGB Composites cannot be modified in this way
+    (i.e. you can't replace "VIS006" by "natural_color").  To get a parallax
+    corrected RGB composite, create a new composite where each input has the
+    modifier applied.  The parallax calculation should only occur once, because
+    calculations are happening via dask and dask should reuse the calculation.
+    """
+
+    def __call__(self, projectables, optional_datasets=None, **info):
+        """Apply parallax correction.
+
+        The argument ``projectables`` needs to contain the dataset to be
+        projected and the height to use for the correction.
+        """
+        (to_be_corrected, cth) = projectables
+        base_area = to_be_corrected.attrs["area"]
+        corrector = self._get_corrector(base_area)
+        plax_corr_area = corrector(cth)
+        res = resample_dataset(
+                to_be_corrected, plax_corr_area,
+                radius_of_influence=50000, fill_value=np.nan)
+        res.attrs["area"] = to_be_corrected.attrs["area"]
+        self.apply_modifier_info(to_be_corrected, res)
+
+        return res
+
+    def _get_corrector(self, base_area):
+        # only pass on those attributes that are arguments by
+        # ParallaxCorrection.__init__
+        sig = inspect.signature(ParallaxCorrection.__init__)
+        kwargs = {}
+        for k in sig.parameters.keys() & self.attrs.keys():
+            kwargs[k] = self.attrs[k]
+        corrector = ParallaxCorrection(base_area, **kwargs)
+        return corrector
+
+
+def _get_satpos_from_cth(cth_dataset):
+    """Obtain satellite position from CTH dataset, height in metre.
+
+    From a CTH dataset, obtain the satellite position lon, lat, altitude/m,
+    either directly from orbital parameters, or, when missing, from the
+    platform name using pyorbital and skyfield.
+    """
+    try:
+        (sat_lon, sat_lat, sat_alt_km) = get_satpos(cth_dataset)
+    except KeyError:
+        logger.warning(
+                "Orbital parameters missing from metadata. "
+                "Calculating from TLE using skyfield and astropy.")
+        (sat_lon, sat_lat, sat_alt_km) = _get_satpos_from_platform_name(cth_dataset)
+    return (sat_lon, sat_lat, sat_alt_km * 1000)
+
+
+def _get_satpos_from_platform_name(cth_dataset):
```

Comment:
I think this should be integrated to `satpy.utils.get_satpos` somehow.

Reply:
Fixed in 26564f5fd7b2bf70eec0c39c769b90a1f0ea4a46